hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/DriverManagerConnectionProviderImpl.java

```diff
@@ -168,11 +168,17 @@ private static Driver loadDriverIfPossible(String driverClassName, ServiceRegist
 
 	@Override
 	public Connection getConnection() throws SQLException {
+		if ( state == null ) {
```

Comment:
I wasn't sure about this one. The Agroal implementation returns a `null` connection but that really doesn't sound like a good idea.

I'll let you judge of that, I can adjust (or even drop these 2 checks if you think they are useless).

Reply:
This seems sensible, thanks.